v-next/hardhat/package.json

```diff
@@ -35,8 +35,8 @@
     "lint:fix": "pnpm prettier --write && pnpm eslint --fix",
     "eslint": "eslint \"src/**/*.ts\" \"test/**/*.ts\"",
     "prettier": "prettier \"**/*.{ts,js,md,json}\"",
-    "test": "node --import tsx/esm --test --test-reporter=@nomicfoundation/hardhat-node-test-reporter \"test/**/*.ts\"",
-    "test:only": "node --import tsx/esm --test --test-only --test-reporter=@nomicfoundation/hardhat-node-test-reporter \"test/**/*.ts\"",
+    "test": "node --import tsx/esm --test --test-reporter=@nomicfoundation/hardhat-node-test-reporter \"test/src/**/*.ts\"",
+    "test:only": "node --import tsx/esm --test --test-only --test-reporter=@nomicfoundation/hardhat-node-test-reporter \"test/src/**/*.ts\"",
```

Comment:
I don't understand this change. Can you shed some light here? If we need this, I'll modify the github actions check that validates the npm scripts.

Reply:
Yes, so if you look at the new fixture project added with this PR, you'll see there are script files that import the HRE. The HRE runs initialization steps (specifically config resolution steps) when it gets imported. The current test script uses the glob `"test/**/*.ts"`, which makes node's test runner load all typescript files in the `test` directory. This was fine when we only had tests in the `test` directory, but now it causes both the `fixture-projects` as well as `helpers` to also get loaded by the test runner. This has actually been happening for a while, but is only now becoming a problem because when the test runner loads the new fixture project, it causes the HRE instances to be immediately initialized outside the scope of a test that might be using them. This causes an error since the config loading process expectedly can't find a hardhat config file (since they're being unintentionally run outside the scope of the `useFixtureProject` helpers)

Reply:
This change moves the actual test files into `test/src` so we can specifically pass that glob to the node test runner. This means we can put anything else that we want/need to inside `test/*` and not worry about it being loaded unintentionally by the node test runner.